streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -1113,6 +1116,16 @@ private void commitOffsetsOrTransaction(final Map<Task, Map<TopicPartition, Offs
         }
     }
 
+    private void updateTaskMetadata(final Map<TopicPartition, OffsetAndMetadata> allOffsets) {
+        for (final Task task: tasks.activeTasks()) {
+            for (final TopicPartition topicPartition: task.inputPartitions()) {
+                if (allOffsets.containsKey(topicPartition)) {
+                    task.getCommittedOffsets().put(topicPartition, allOffsets.get(topicPartition).offset());
```

Comment:
I see why we need to return a modifiable map here. My suggestion is to separate the setter and getter so that when we return a map for `get` call, it is a deep-copy and doesn't attach to the thread anymore. The benefit is that application level won't accidentally mutate the thread state, thus providing better encapsulation IMHO.